base-compat.cabal

```diff
@@ -35,18 +35,35 @@ library
   exposed-modules:
       Prelude.Compat
       Control.Applicative.Compat
+      Control.Concurrent.MVar.Compat
       Control.Exception.Compat
       Control.Monad.Compat
       Data.Bool.Compat
       Data.Either.Compat
       Data.Foldable.Compat
+      Data.Function.Compat
       Data.Functor.Compat
+      Data.List.Compat
       Data.Monoid.Compat
+      Data.Ord.Compat
       Data.Traversable.Compat
+      Data.Version.Compat
+      Debug.Trace.Compat
+      Foreign.Marshal.Alloc.Compat
+      Foreign.Marshal.Array.Compat
+      Foreign.Storable.Compat
+      System.Console.GetOpt.Compat
       System.Environment.Compat
       System.Exit.Compat
       Text.Read.Compat
 
+  if impl(ghc >= 7.2)
+    exposed-modules:
+      GHC.Generics.Compat
+    if impl(ghc < 7.6)
```

Comment:
The only catch is that the `GHC.Generics` module is only defined on GHC 7.2 and later, so I'd have to add some more CPP pragmas to ensure that `GHC.Generics.Compat` doesn't export anything on GHC 7.0.